homeassistant/components/bryant_evolution/__init__.py

```diff
@@ -0,0 +1,38 @@
+"""The Bryant Evolution integration."""
+
+from __future__ import annotations
+
+import logging
+
+from evolutionhttp import BryantEvolutionClient
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, Platform
+from homeassistant.core import HomeAssistant
+
+from .const import CONF_SYSTEM_ID, CONF_ZONE_ID
+
+PLATFORMS: list[Platform] = [Platform.CLIMATE]
+
+type BryantEvolutionConfigEntry = ConfigEntry[BryantEvolutionClient]  # noqa: F821
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: BryantEvolutionConfigEntry
+) -> bool:
+    """Set up Bryant Evolution from a config entry."""
+    if not hasattr(entry, "runtime_data") or entry.runtime_data is None:
```

Comment:
I provide a runtime_data that fakes the client in unit tests. If I don't have this check, it'll be unconditionally overwritten. When I just test "runtime_data is None," I get a no-such-attribute error.

Apologies if I am missing something obvious. Python is not my native language. :)

Reply:
Ah, you're doing a dependency injection approach. Our tests are usually quite end to end, we use the public APIs to execute actions and check the result of them in the stores and registries, so we rather want you to let HA set up the integration and mock the library

Reply:
Could I ask you to take a look at mock_evolution_entry in test_climate.py? This is how I am setting up the integration. Did you want something more end-to-end? Could you suggest if so?

Reply:
In particular, would you be happy with the existing code for integration setup in the test if I mocked the client creation method from the evolutionhttp library, rather than injecting the client as I currently do?

Reply:
Made that change